src/js/itemset.js

```diff
@@ -185,11 +185,11 @@ export default {
                                     const language = label.getAttribute( 'lang' );
                                     const type = label.nodeName;
                                     const src = label.src;
-                                    const text = label.textContent;
+                                    const contentNodes = [ ...label.childNodes ];
```

Comment:
Instead of only adding textContent, all nodes will now be added to the itemset (including `<span class="or-output">` nodes)

Reply:
I was curious about the significance of this change. So I rolled it back (keeping the cache changes), and the difference I see is:

1. The cache changes still partially fix the issue as described in #844, but
2. The labels are not updated until a new repeat is added (also observed on master)

Which in turn made me curious about how each caching change affects behavior, so I:

- Rolled back [both](https://github.com/enketo/enketo-core/pull/846/files#diff-608210f16bb20aa58380a2606ca9a842e85e4634d9e3d03347845848b9bc0f8cR534) [changes](https://github.com/enketo/enketo-core/pull/846/files#diff-608210f16bb20aa58380a2606ca9a842e85e4634d9e3d03347845848b9bc0f8cR550), and indeed I can reproduce #844 again.
- Rolled back only the first change, and the fix still works (I'll add another comment there suggesting it be rolled back, but this was all one thought process so that's why I'm noting it here).
- Rolled back only the second change, and again only see labels update when adding a new repeat.

Reply:
Ah nice find! Indeed, I misread the code and the second cache change is not required!